### PR TITLE
Allow to ignore overloaded methods for the complex interface rule (#5165)

### LINKED ISSUE #5165: ComplexInterface: do not count method overloads
The rule or changes you're looking for might have already been suggested!
Please search in the [issues](https://github.com/detekt/detekt/issues) before creating a new one.

## Expected Behavior of the rule
It will be nice if `ComplexInterface` rule will have a `skipOverloadMethods` option.

If the option is on, it will count overloaded methods as 1.

### PR DESCRIPTION
**Pull request by pt2121**
Closes #5165

ComplexInterface: do not count method overloads

### CODE REVIEW

File: detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterface.kt [hunk 1]
```diff
@@ -51,6 +52,9 @@ class ComplexInterface(
     @Configuration("whether private declarations should be included")
     private val includePrivateDeclarations: Boolean by config(defaultValue = false)
 
+    @Configuration("ignore overloaded methods - only count once")
+    private val ignoreOverloaded: Boolean by config(defaultValue = false)
```
Comment (schalkms) [inline detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterface.kt:56]: to be consistent with all other detekt rules

```suggestion
    @Configuration("ignore overridden methods - only count once")
    private val ignoreOverridden: Boolean by config(defaultValue = false)
```

File: detekt-rules-complexity/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterfaceSpec.kt [hunk 2]
```diff
@@ -141,6 +144,57 @@ class ComplexInterfaceSpec {
...
+
+            @Test
+            fun `does not report simple interface with ignoreOverloaded`() {
+                val rule = ComplexInterface(ignoreOverloadedConfig)
+                assertThat(rule.compileAndLint(code)).isEmpty()
+            }
+
+            @Test
+            fun `reports complex interface with extension methods with a different receiver`() {
+                val code = """
```
Comment (schalkms) [inline detekt-rules-complexity/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterfaceSpec.kt:172]: `code` is shadowed here twice. Please try to restructure the tests to mitigate this Kotlin warning.

**Review by schalkms — COMMENTED** [review]
Thank you for this nice first PR! I like it.

Below, you find two comments regarding naming and style.
Then, I'd say LGTM.

File: detekt-rules-complexity/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterfaceSpec.kt (hunk 2 above)
Comment (pt2121) [reply detekt-rules-complexity/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterfaceSpec.kt:172]: Thanks for the feedback. Somehow, my IDE doesn't show the warning. I'll look into it :+1:

File: detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterface.kt (hunk 1 above)
Comment (pt2121) [reply detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterface.kt:56]: These methods are overloaded - not overridden, right? :thinking:
Let me recheck.

Comment (schalkms) [reply detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ComplexInterface.kt:56]: I got mislead by this comment. The naming is fine. We can keep it.

https://github.com/detekt/detekt/issues/5165#issuecomment-1200915187

**Review by schalkms — APPROVED** [review]

**Comment by pt2121** [comment]
Thanks for reviewing and merging it

**Pull request merged**